src/wireguard.rs

```diff
@@ -0,0 +1,305 @@
+
+
+elevated permisions / authenticated user: 
+
+     set private key                 params: privkey, id      - Sets a private key. If an ID is provided -> overwrite. Returns the internal private key UID
+     generate keypair                params: <none>           - Generates a keypair, saves the private key and returns a public key, as well as the internal private key UID
+     delete private key              params  id               - Deletes a private key from the device with a given ID 
+
+     set user config                 params: configUID, value - Change the config. Probably needed to change the behaviour stated in `get AEAD`.
+
+
+___________
+
+Perhaps not interesting for NPX: 
+
+    set time                                     - No RTC available on the STM32, that was capable of incrementing a day/month/year. 
+                                                  Restricted to time exclusively. Solution: transmit the current epoch and extract the date.
+
+    get time                                     - get the device time. Mostly to check, whether a new timestamp is needed.
+
+
+
+**/
+
+use log::{debug, info};
+use serde::{Deserialize, Serialize};
+use trussed::{api::reply::Delete, consts, syscall, try_syscall, types::Message};
+use trussed::{ByteBuf, types::{Mechanism, /*SignatureSerialization, StorageAttributes,*/ StorageLocation}};
+
+use crate::Result;
+
+
+// Some constants
+
+const SIZE_CK: usize = 32;
+const SIZE_HS: usize = 32;
+const SIZE_PUBKEY: usize = 32;
+const SIZE_PRIVKEY: usize = 32;
+
+
+#[allow(missing_docs)]
+// core wireguard app
+pub struct Wireguard <T : trussed::Client>
+{
+    trussed: T,
+}
+
+/*
+    Commands / Requests
+ */
+
+ /*
+  get AEAD                        params: pubkey, C, H     - The most important function, since the AEAD is necessary to perform the NOISE handshake.
+                                                              - Transmitts the peer's public key, the chained key C and chained Hash 'H' to the device. Returns AEAD.
+                                                              - Requires: User action (e.g. Button interaction) on first use. Perhaps additionaly every X minutes (configurable)
+ */
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct GetAead 
+ {
+     pub pubkey: [u8;SIZE_PUBKEY],
+     pub c : [u8; SIZE_CK],
+     pub h : [u8; SIZE_HS],
+     pub key_id : u32
+ }
+
+ //  Authenticate                    params: PIN/pass         - Unlocks the device to use it. 
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct Unlock
+ {
+     pub pin: String, // pin code to unlock the device
+ }
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct RegisterKeyPair
+ {
+     pub privkey : [u8; SIZE_PRIVKEY],
+     pub pubkey : [u8; SIZE_PUBKEY],
+     pub label : String
```

Comment:
Probably not wrong to start using `heapless::String<CAPACITY>` soon-ish, as you won't have `alloc::String` on the device.